cirq/api/google/v2/device.proto

```diff
@@ -0,0 +1,96 @@
+syntax = "proto3";
+
+package cirq.api.google.v2;
+
+option java_package = "com.google.cirq.api.google.v2";
+option java_outer_classname = "CalibrationProto";
+option java_multiple_files = true;
+
+// This contains information about a device that includes the
+// qubits on the device, supported gates, connections, and timing.
+// This message specifies information that is needed when sending a
+// Program message to the device.
+message DeviceSpecification {
+
+  // A list of allowed gatesets for programs submitted to this processor
+  // Language.gate_set should be one of these values to be valid.
+  repeated GateSet valid_gate_sets = 1;
+
+  // A list of allowed ids for qubits within the Program.
+  // Any programs with ids not in this list will be rejected.
+  // If empty, all qubit values are allowed (e.g. in a simulator)
+  repeated string valid_qubits = 2;
+
+  // A list of allowed targets for multi-qubit gates.
+  // For a typical grid layout, this will be a list of adjacent pairs
+  repeated Target valid_targets = 3;
```

Comment:
for gates that are not symmetric, does the order matter?

Reply:
I have changed this to use a TargetList and have made is_symmetric an attribute on the TargetList.